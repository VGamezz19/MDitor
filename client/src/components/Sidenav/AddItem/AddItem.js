import React, { Component } from 'react'
import IconButton from 'material-ui/IconButton'
import Delete from 'material-ui/svg-icons/content/clear'
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';

class AddItem extends Component {
    constructor(props) {
        super(props)

        this.state = {
            value: ''
        }
    }

    componentDidMount() {
        document.addEventListener('keydown', this._handleEscKey, false);
    }

    componentWillUnmount() {
        document.removeEventListener('keydown', this._handleEscKey, false);
    }

    _handleEscKey = event => event.keyCode === 27 ? this.handlerCancel() : false;

    handlerChangeInput = value => this.setState({ value })

    handlerCancel = () => {
        const { onCancel } = this.props

        if (onCancel) {
            onCancel()
        }
    }

    handlerSubmit = (value) => {
        const { onSubmit } = this.props

        if (onSubmit) {
            onSubmit(value)
        }
    }

    render() {
        const { itemType, className, inputType } = this.props
        const { value } = this.state

        return (
            <form
                className={`format ${itemType} ${className}`}
                onSubmit={(event) => {
                    event.preventDefault()
                    this.handlerSubmit(value)
                }}>
                <input
                    className='inputItem'
                    autoFocus
                    type={inputType}
                    value={value}
                    onChange={(event) => {
                        event.preventDefault();
                        this.handlerChangeInput(event.target.value)
                    }} />

                <MuiThemeProvider>
                    <IconButton className='button-delete' onClick={this.handlerCancel}>
                        <Delete />
                    </IconButton>
                </MuiThemeProvider>

            </form>
        )
    }
}
export default AddItem