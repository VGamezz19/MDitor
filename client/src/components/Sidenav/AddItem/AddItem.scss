@import '../../../sass/_variables';
@import '../../../sass/_extends';
.format {
    @extend %flex-in-column;
    @extend %rectangle-container;
    flex-direction: row !important;
    justify-content: space-between;
    .inputItem {
        border-bottom: 1px solid $main-border-color !important;
    }
    .inputItem:focus {
        border-bottom: 1px solid $main-border-color !important;
        box-shadow: 0 1px 0 0 $main-border-color !important;
    }
    .button-delete:hover {
        div {
            svg {
                color: $delete-color !important;
            }
        }
    }
}

.file {
    background-color: $file-item-color;
    padding: 0 1em 0 2em !important
}

.folder {
    background-color: $folder-item-color!important;
    padding: 0px 1em 0 4.4em;
}