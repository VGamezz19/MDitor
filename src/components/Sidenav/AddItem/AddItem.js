import React, { Component } from 'react'
import IconButton from 'material-ui/IconButton';
import Delete from 'material-ui/svg-icons/content/clear';
class AddItem extends Component {
    constructor(props) {
        super(props)

        this.state = {
            inputValue: ''
        }
    }

    handlerChangeInput = inputValue => this.setState({ inputValue })

    handlerSubmit = () => this.props.onSubmit(this.state.inputValue)

    handlerCancel = () => this.props.onCancel()

    render() {
        return (
            <form
                className={`format ${this.props.itemType} ${this.props.className}`}
                onSubmit={(event) => {
                    event.preventDefault()
                    this.handlerSubmit()
                }}>
                <input
                    className='inputItem'
                    autoFocus
                    type={this.props.inputType}
                    value={this.state.value}
                    onChange={(event) => {
                        event.preventDefault();
                        this.handlerChangeInput(event.target.value)
                    }} />

                <IconButton className='button-delete' onClick={this.handlerCancel}>
                    <Delete />
                </IconButton>
            </form>
        )
    }
}
export default AddItem