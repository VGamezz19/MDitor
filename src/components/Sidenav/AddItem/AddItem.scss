@import '../../../sass/_variables';
@import '../../../sass/_extends';

.format {
    @extend %flex-in-column;
    @extend %rectangle-container;
    
}

.file {

}

.folder {
    background-color: $side-nav-main-color !important;
    padding: 0px 3.4em;
}