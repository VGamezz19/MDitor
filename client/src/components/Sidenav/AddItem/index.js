import React from 'react'
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';

import AddItem from './AddItem'
import './AddItem.scss'

export default AddItem


/**
 * Exporting Cosmos Fixture
 */
function AddItemFixture() {
    return <MuiThemeProvider>
            <AddItem/>
        </MuiThemeProvider>
}

export { AddItemFixture }