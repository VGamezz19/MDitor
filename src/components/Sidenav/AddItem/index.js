import AddItem from './AddItem'
import './AddItem.scss'

export default AddItem