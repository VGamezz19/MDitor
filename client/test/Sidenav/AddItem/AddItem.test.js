import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';

import AddItem from '../../../src/components/Sidenav/AddItem';

it('Main aplication rendering', () => {

    const div = document.createElement('div');

    const wrapper = <AddItem
        itemType='file'
        inputType='text'
        onSubmit={() => console.log("submit")}
        onCancel={() => console.log("cancel")} />

    ReactDOM.render(wrapper, div);

    ReactDOM.unmountComponentAtNode(div);
});

it('Should create and Match same Snapshot', () => {

    const wrapper = <AddItem
        itemType='file'
        inputType='text'
        onSubmit={() => console.log("submit")}
        onCancel={() => console.log("cancel")} />

    const component = renderer.create(wrapper);

    let tree = component.toJSON();

    expect(tree).toMatchSnapshot();
});