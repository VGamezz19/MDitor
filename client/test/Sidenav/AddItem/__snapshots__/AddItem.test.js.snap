// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should create and Match same Snapshot 1`] = `
<form
  className="format file undefined"
  onSubmit={[Function]}
>
  <input
    autoFocus={true}
    className="inputItem"
    onChange={[Function]}
    type="text"
    value=""
  />
  <button
    className="button-delete"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onMouseOut={[Function]}
    onTouchStart={[Function]}
    style={
      Object {
        "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
        "background": "none",
        "border": 10,
        "boxSizing": "border-box",
        "cursor": "pointer",
        "display": "inline-block",
        "fontFamily": "Roboto, sans-serif",
        "fontSize": 0,
        "fontWeight": "inherit",
        "height": 48,
        "margin": 0,
        "muiPrepared": true,
        "outline": "none",
        "overflow": "visible",
        "padding": 12,
        "position": "relative",
        "textDecoration": "none",
        "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
        "verticalAlign": null,
        "width": 48,
      }
    }
    tabIndex={0}
    type="button"
  >
    <div
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
    >
      <svg
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        style={
          Object {
            "color": "rgba(0, 0, 0, 0.87)",
            "display": "inline-block",
            "fill": "currentColor",
            "height": 24,
            "muiPrepared": true,
            "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
            "userSelect": "none",
            "width": 24,
          }
        }
        viewBox="0 0 24 24"
      >
        <path
          d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
        />
      </svg>
    </div>
  </button>
</form>
`;
